src/App.js

```diff
@@ -1,38 +1,54 @@
+import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
+import { NOT_FOUND_KEY, WEATHER_KEY, WORK_KEY } from './constants';
+import WeatherDataContextProvider from './contexts/WeatherDataContext';
+import WeatherInputContextProvider from './contexts/WeatherInputContext';
+import WorkInputContextProvider from './contexts/WorkInputContext';
+import WorkDataContextProvider from './contexts/WorkDataContext';
 import Header from './components/Header/Header';
-import DataContextProvider from './contexts/DataContext';
-import InputContextProvider from './contexts/InputContext';
-import WorkContextProvider from './contexts/WorkContext';
+import Home from './components/Home/Home';
 import WeatherChart from './components/Charts/WeatherChart';
 import WeatherTable from './components/Tables/WeatherTable';
 import WeatherButton from './components/WeatherInput/WeatherButton';
 import WorkForm from './components/WorkInput/WorkForm';
-import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
 import WorkTable from './components/Tables/WorkTable';
+import WorkDetails from './components/Modals/WorkDetails';
+import NotFound from './components/NotFound';
 
 function App() {
   return (
-    <DataContextProvider>
-      <InputContextProvider>
-        <Router>
-          <Header />
-          <div className='layout'>
-            <Switch>
-              <Route exact path={'/'}>
-                <WeatherTable />
-                <WeatherButton />
-                <WeatherChart />
-              </Route>
-              <Route exact path={'/work'}>
-                <WorkContextProvider>
-                  <WorkForm />
-                  <WorkTable />
-                </WorkContextProvider>
-              </Route>
-            </Switch>
-          </div>
-        </Router>
-      </InputContextProvider>
-    </DataContextProvider>
+    <WeatherDataContextProvider>
+      <WeatherInputContextProvider>
+        <WorkDataContextProvider>
+          <WorkInputContextProvider>
+            <Router>
+              <Header />
+              <div className='layout'>
+                <Switch>
+                  <Route exact path={'/'}>
+                    <Home />
+                  </Route>
+                  <Route exact path={WEATHER_KEY}>
+                    <WeatherTable />
+                    <WeatherButton />
+                    <WeatherChart />
+                  </Route>
+                  <Route exact path={WORK_KEY}>
+                    <WorkForm />
+                    <WorkTable />
+                  </Route>
+                  <Route exact path='/work-details/:id'>
+                    <WorkDetails />
+                  </Route>
+                  <Route>
+                    <NotFound path={NOT_FOUND_KEY} />
```

Comment:
`path` is something Route takes and there's no particular route for the fallback page so the route above this component is correct.  However, why are you passing in a path to the NotFound component you created?